freon@UshakovVasilii_Github.yahoo.com/smartctlUtil.js

```diff
@@ -1,13 +1,13 @@
-const GLib = imports.gi.GLib;
-
-const Me = imports.misc.extensionUtils.getCurrentExtension();
+import GLib from 'gi://GLib';
 const ByteArray = imports.byteArray;
+
 function getSmartData (argv){
     const smartctl = GLib.find_program_in_path('smartctl')
     return JSON.parse(ByteArray.toString( GLib.spawn_command_line_sync(`'${smartctl}' ${argv} -j`)[1] ))
```

Comment:
You can switch to `TextDecoder` and drop the `ByteArray` import
```suggestion

function getSmartData (argv){
    const smartctl = GLib.find_program_in_path('smartctl')
    return JSON.parse(new TextDecoder().decode( GLib.spawn_command_line_sync(`'${smartctl}' ${argv} -j`)[1] ))
```

Reply:
I didn't know about TextDecoder, thanks for the suggestion. I've found some additional places where toString() was called directly on a ByteArray and also replaced those.